src/Uno.Extensions.Navigation.UI/Dispatcher.cs

```diff
@@ -33,5 +33,15 @@ public Dispatcher(FrameworkElement element)
 
 	/// <inheritdoc />
 	public async ValueTask<TResult> ExecuteAsync<TResult>(AsyncFunc<TResult> func, CancellationToken cancellation)
-		=> await _dispatcher.ExecuteAsync(func, cancellation);
+	{
+		if (!PlatformHelper.IsWebAssembly && // Assume for web that we always call executeasync on dispatcher
```

Comment:
You'll need to check for explicit Wasm threading support to be properly compatible.